packages/svelte/src/internal/client/reactivity/sources.js

```diff
@@ -32,27 +34,37 @@ let inspect_effects = new Set();
 /**
  * @template V
  * @param {V} v
+ * @param {boolean} [skip_derived_source]
  * @returns {Source<V>}
  */
 /*#__NO_SIDE_EFFECTS__*/
-export function source(v) {
-	return {
+export function source(v, skip_derived_source = false) {
```

Comment:
I tend to think it's better to avoid negatives in variable names (even though a few other `skip_` things have slipped in) since you have to mentally invert stuff to understand what's happening. At first I thought replacing `if (!skip_derived_source ...)` with `if (track_owner ...)` might be clearer...

```suggestion
export function source(v, track_owner = true) {
```

...but then it occurred to me that we could even do this...

```suggestion
export function source(v, owner = current_reaction) {
```

...and then just do this below:

```js
if (owner !== null && (owner.f & DERIVED) !== 0) {...}
```

One less thing to check. Thoughts?

Reply:
Good idea

Reply:
made that change